opta/core/cloud_client.py

```diff
@@ -0,0 +1,24 @@
+from typing import TYPE_CHECKING, Optional
+
+if TYPE_CHECKING:
+    from opta.layer import Layer, StructuredConfig
+
+
+class CloudClient:
```

Comment:
```suggestion
class CloudClient(ABC):
```

- Inheriting from `abc.ABC` (abstract base class) prevents CloudClient from being instantiated directly.
- Funnily enough, I had almost done the same thing for https://github.com/run-x/opta/pull/483 but realized it wasn't necessary for that PR.